entrypoint.sh

```diff
@@ -35,28 +35,28 @@ function convertToKeyValuePairs() {
 }
 
 function getRunningWorkflowIds() {
-  jq ".workflow_runs | .[] | select(.head_branch==\"${branch}\" and .head_repository.full_name==\"${repo}\" and .status==\"in_progress\" or .status==\"queued\" or .status== \"waiting\") | .id " | grep -v "${GITHUB_RUN_ID}" || :
+  jq ".workflow_runs | .[] | select(.head_branch==\"${branch?}\" and .head_repository.full_name==\"${repo?}\" and .status==\"in_progress\" or .status==\"queued\" or .status== \"waiting\") | .id " | grep -v "${GITHUB_RUN_ID}" || :
 }
 
 function exportAll() {
   for var in $1; do
-    export $var
+    export "${var?}"
   done
 }
 
 # extract meta information for current workflow run
 exportAll "$(curl -s "${GITHUB_API}/repos/${GITHUB_REPOSITORY}/actions/runs/${GITHUB_RUN_ID}" -H "${auth_header}" -H "${accept_header}" | extractMetaInformation | convertToKeyValuePairs)"
-echo "workflow id: $workflow_id"
-echo "branch: $branch"
-echo "repo: $repo"
+echo "workflow id: ${workflow_id?}"
+echo "branch: ${branch?}"
+echo "repo: ${repo?}"
```

Comment:
According to shellcheck this way of creating variables is not good. The error message is that we use variables which are not declared. Technically speaking they are not declared at compile time, but will be declared at run time by the `exportAll` function.

Currently we use the function `convertToKeyValuePairs` which resolves the json object generated by `extractMetaInformation` to list of valid export strings. That list is then exported as variables.

Do we want to keep that behavior? I admit that this implicit way of declaring the variables can be a little confusing. Making it explicit could increase the readability at the cost of code compactness. What do you thing?

As of right now I just muted all error messages concerning this issue. If we want to change the way we generate our variables we should do that as a separate issue.

Reply:
Hi @mustaphazorgati yes, I wondered as well how we should fix this - I think code readability in a script like this is more important than compactness so I would vote for the more verbose approach and not muting this particular error.

Reply:
-> #26